Simplified/AxisService/NYPLRSACypher.swift

```diff
@@ -126,52 +128,95 @@ struct NYPLRSACypher: NYPLRSACryptographing {
     
     self.publicKey = publicKey.base64EncodedString()
     self.privateKey = privateKey.base64EncodedString()
-    self.modulus = self.publicKey.replacingOccurrences(of: "/", with: "-")
-    // we're creating a key with 2048 bits. Exponent for that is AQAB.
-    self.exponent = "AQAB"
-  }
-  
-  func encryptText(_ message: String) -> String? {
-    let _messageData = message.data(using: .utf8)
     
-    guard let messageData = _messageData else {
+    guard
+      let (mod, exp) = NYPLRSACypher.parsePublicSecKey(publicKey: self.publicKeySec)
+    else {
       NYPLRSACypher
-        .logFailedEncryptionError(NYPLRSACypher.stringToDataConversionFailure)
-      return nil
-    }
-    guard let encryptedData = SecKeyCreateEncryptedData(publicKeySec,
-                                                        .rsaEncryptionOAEPSHA1,
-                                                        messageData as CFData,
-                                                        nil)
-      else {
-        NYPLRSACypher
-          .logFailedEncryptionError(NYPLRSACypher.nilEncryptedDataFailure)
+        .logFailedInitializationError(
+          "Failed to get exponent and or modulus for public key")
         return nil
     }
     
-    return (encryptedData as Data).base64EncodedString()
+    let modulus = mod.base64EncodedString()
+    guard modulus.count >= 342 else {
```

Comment:
why 342? can you add a comment about this hardcoded constant?

Reply:
Done